util/src/length_prefix_encoding/encoder.rs

```diff
@@ -92,6 +92,90 @@ pub struct WriteToIoReturn {
 }
 
 #[derive(Clone, Copy, Debug)]
+/// Module description goes here (skipped since it already exists in #479)
+///
+/// # Examples
+///
+///  Unfinished/proof-of-concept versions below - don't try this at home
+///
+/// **TODOs**:
+///
+/// * Actually write to a sink, if that's desirable/integration tests don't suffice
```

Comment:
I think writing to a dummy sink (memory buffer, I think you can use std::io::Cursor along with static array) would be a great demo!